web/src/read_receipts.ts

```diff
@@ -12,12 +14,20 @@ import * as modals from "./modals";
 import * as people from "./people";
 import * as ui_report from "./ui_report";
 
-export function show_user_list(message_id) {
+const read_receipts_api_response_schema = z.object({
+    user_ids: z.array(z.number()),
+});
+
+export function show_user_list(message_id: number): void {
     $("body").append(render_read_receipts_modal());
     modals.open("read_receipts_modal", {
         autoremove: true,
         on_show() {
             const message = message_store.get(message_id);
+            if (message === undefined) {
+                blueslip.error("message is undefined", {message});
+                return;
+            }
```

Comment:
You should pass the `message_id`, not the message that we know is undefined, here.

I think it might be impossible for this to fail, though, so an assert might be more appropriate.

Reply:
Such a race isn’t possible in single-threaded JavaScript; `modals.open` synchronously invokes its `on_show` callback before returning.

This would previously have thrown on the next line accessing `message.sender_email`, and I don’t see any matching reports in Sentry. We can just `assert`.